python/ray/serve/controller.py

```diff
@@ -309,6 +322,8 @@ async def run_control_loop(self) -> None:
             except Exception:
                 logger.exception("Exception updating application state.")
 
+            self._update_http_proxy_draining_flag()
```

Comment:
Why is this not part of regular `http_state.update()`? We should centralize all state update logic for the proxies there, otherwise the interactions are harder to reason about.

I would suggest instead making `http_state.update()` take the "active node map" as an argument and handling the logic there.

Reply:
Yea, I was thinking that or doing what's in this PR. I thought `update()` is a standard method that we shouldn't change the interface, so do it the way it is. Will just pass in active nodes as argument

Reply:
Ended up using the long pull to pass the active node set changes. Will no longer need to call this separately :)